var/spack/repos/builtin/packages/cairo/package.py

```diff
@@ -55,25 +74,52 @@ class Cairo(AutotoolsPackage):
     depends_on("glib")
     depends_on("pixman@0.36.0:", when="@1.17.2:")
     depends_on("pixman")
-    depends_on("automake", type="build")
-    depends_on("autoconf", type="build")
-    depends_on("libtool", type="build")
-    depends_on("m4", type="build")
     depends_on("freetype build_system=autotools", when="+ft")
+    # Require freetype with FT_Color
+    # https://gitlab.freedesktop.org/cairo/cairo/-/issues/792
+    depends_on("freetype@2.10:", when="@1.18.0: +ft")
     depends_on("pkgconfig", type="build")
     depends_on("fontconfig@2.10.91:", when="+fc")  # Require newer version of fontconfig.
     depends_on("which", type="build")
+    depends_on("zlib", when="+pdf")
 
     conflicts("+png", when="platform=darwin")
     conflicts("+svg", when="platform=darwin")
     conflicts("+shared~pic")
 
     # patch from https://gitlab.freedesktop.org/cairo/cairo/issues/346
     patch("fontconfig.patch", when="@1.16.0:1.17.2")
-    # Don't regenerate docs to avoid a dependency on gtk-doc
-    patch("disable-gtk-docs.patch", when="^autoconf@2.70:")
+    # Patch autogen.sh to not regenerate docs to avoid a dependency on gtk-doc
+    patch("disable-gtk-docs.patch", when="build_system=autotools ^autoconf@2.70:")
+
+    def check(self):
+        """The checks are only for the cairo devs: They write others shouldn't bother"""
+        pass
+
 
-    def autoreconf(self, spec, prefix):
+class MesonBuilder(spack.build_systems.meson.MesonBuilder):
+    def meson_args(self):
```

Comment:
```suggestion
    def setup_build_environment(self, env):
        if self.spec.satisfies("+ft ^freetype~shared"):
            pkgconf = which("pkg-config")
            env.append_flags("CFLAGS", pkgconf("--libs", "--static", "freetype2", output=str))

    def meson_args(self):
```

Reply:
or
```suggestion
    @when("+ft ^freetype~shared")
    def setup_build_environment(self, env):
        pkgconf = which("pkg-config")
        env.append_flags("CFLAGS", pkgconf("--libs", "--static", "freetype2", output=str))

    def meson_args(self):
```

Reply:
Yeah, sorry for the delay on this. I started testing with `freetype ~shared` and now I'm not able to build fontconfig without those changes either. I'm not sure adding all these extra cflags to packages downstream from freetype is the right thing to do. It seems something is missing in freetype's pc files: they should get picked up by spack, and the ldflags should include the `-lpng` library.